From ae01ccbc0fe2842ce4e022495ba1797acebcc8da Mon Sep 17 00:00:00 2001
From: Kapileshwar Singh <kapileshwar.singh@arm.com>
Date: Fri, 12 Jun 2015 22:13:32 +0100
Subject: [PATCH 13/38] WIP: arm64: ForkMigration: SD_FORK_HIGHEST_CAP at
 cluster level

Add SD_FORK_HIGHEST_CAP to arm64 DIE level domain in order to
facilitate ForkMigration for arm64.

Change-Id: I88e14a1501875457e3305ccdb1b3ab7c4f97c8d9
Signed-off-by: Kapileshwar Singh <kapileshwar.singh@arm.com>
---
 arch/arm64/kernel/topology.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/kernel/topology.c b/arch/arm64/kernel/topology.c
index 510b7f1..177ad64 100644
--- a/arch/arm64/kernel/topology.c
+++ b/arch/arm64/kernel/topology.c
@@ -273,11 +273,16 @@ static inline int cpu_corepower_flags(void)
 	       SD_SHARE_CAP_STATES;
 }
 
+static inline int arm64_die_flags(void)
+{
+	return SD_FORK_HIGHEST_CAP;
+}
+
 static struct sched_domain_topology_level arm64_topology[] = {
 #ifdef CONFIG_SCHED_MC
 	{ cpu_coregroup_mask, cpu_corepower_flags, cpu_core_energy, SD_INIT_NAME(MC) },
 #endif
-	{ cpu_cpu_mask, 0, cpu_cluster_energy, SD_INIT_NAME(DIE) },
+	{ cpu_cpu_mask, arm64_die_flags, cpu_cluster_energy, SD_INIT_NAME(DIE) },
 	{ NULL, },
 };
 
-- 
1.9.1

